src/olympia/discovery/views.py

```diff
@@ -22,6 +22,11 @@ def get_queryset(self):
         addons = self.queryset.in_bulk(ids)
 
         # Patch items to add addons.
+        result = []
         for item in discopane_items:
-            item.addon = addons[item.addon_id]
-        return discopane_items
+            try:
+                item.addon = addons[item.addon_id]
+                result.append(item)
+            except KeyError:
+                pass
```

Comment:
Maybe log via some statsd counter that can be seen on a dashboard and where we can create alert rules for?